### PR TITLE
Mark notifications as read when snoozing, ignoring, archiving tasks

### LINKED ISSUE #55: Allow muting/clearing relevant notification
The tasks shown are really just issues and pull requests pulled from the [GitHub search API](https://developer.github.com/v3/search/). It would be helpful if the relevant notifications from the [notifications API](https://developer.github.com/v3/activity/notifications/) could be muted or marked as read as you ignore / archive tasks.

I think snoozing is a concept unique to tasks in this app, and nothing should be done with any notification based on a snooze.

Not all tasks shown will have a notification. The user can make filters for issues and PRs that they would not receive notifications for.

### PR DESCRIPTION
**Pull request by cheshire137**
Fixes #55. This will mark any associated notification on GitHub as read when you take an action on a task such as snoozing, archiving, or ignoring it. This way, you won't have to do double work by managing tasks in the snoozer app and also clearing out your notifications on GitHub for the same issues and pull requests you've already dealt with.

It works by fetching notifications first, then searching issues/PRs by the selected filter. It associates notifications to tasks based on their API URL. If the notification is for the same subject (issue/PR) as a task being displayed, then when you archive/ignore/snooze the task, a request is sent to mark the notification as read.

Bonus:
- Adds a new default filter, My Repositories, that shows all issues/PRs opened by others on the user's own repos. I found this useful to keep track of pull requests others were submitting to my personal projects.

### CODE REVIEW

**Comment by cheshire137** [comment]
@probablycorey and @summasmiff, this is ready for review now! 🎉

**Comment by summasmiff** [comment]
@cheshire137, got some failing tests!

**Comment by cheshire137** [comment]
Tests are passing now!

File: src/models/GitHub.js [hunk 1]
```diff
@@ -43,8 +48,18 @@ class GitHub extends Fetcher {
   }
 
   // https://developer.github.com/v3/activity/notifications/#list-your-notifications
-  getNotifications() {
-    return this.get('notifications')
+  getNotifications(sinceDate) {
+    let date = sinceDate
+    if (typeof date === 'undefined') {
+      if (process.env.NODE_ENV === 'test') {
+        date = new Date(Date.UTC(2016, 5, 4)) // June 4, 2016 UTC
```
Comment (cheshire137) [inline src/models/GitHub.js]: "Putting test code in production code makes my ❤️  shiver." - @probablycorey

**Comment by cheshire137** [comment]
I paired on this with @probablycorey a bit to improve function names and make testing getNotifications a bit nicer.

**Pull request merged**